base_agent/task.py

```diff
@@ -21,63 +56,213 @@ class Task(object):
         name (string): Name of the task
         undone (bool): A flag indicating whether the task was undone / reverted
         last_stepped_time (int): Timestamp of last step through the task
-        throttling_tick (int): The threshold beyond which the task will be throttled
         stop_condition (Condition): The condition on which the task will be stopped (by default,
                         this is NeverCondition)
     Examples::
         >>> Task()
     """
 
-    def __init__(self):
-        self.memid = None
+    def __init__(self, agent, task_data={}):
+        self.agent = agent
+        self.run_count = 0
         self.interrupted = False
         self.finished = False
         self.name = None
         self.undone = False
         self.last_stepped_time = None
-        self.throttling_tick = DEFAULT_THROTTLING_TICK
-        self.stop_condition = NeverCondition(None)
+        self.prio = -1
+        self.running = 0
+        self.memid = TaskNode.create(self.agent.memory, self)
+        # TODO put these in memory in a new table?
+        # TODO methods for safely changing these
+        i, s, ru, re = get_default_conditions(task_data, agent, self)
+        self.init_condition = i
+        self.stop_condition = s
+        self.run_condition = ru
+        self.remove_condition = re
+        TripleNode.create(
+            self.agent.memory,
+            subj=self.memid,
+            pred_text="has_name",
+            obj_text=self.__class__.__name__.lower(),
+        )
+        TaskNode(agent.memory, self.memid).update_task(task=self)
+        # TODO if this is a command, put a chat_effect triple
 
-    def step(self, agent):
-        """The actual execution of a single step of the task is defined here."""
-        # todo? make it so something stopped by condition can be resumed?
-        if self.stop_condition.check():
-            self.finished = True
+    @staticmethod
+    def step_wrapper(stepfn):
```

Comment:
this implementation has a bug. If `step(...)` takes arguments, for example https://github.com/facebookresearch/droidlet/blob/67bfa83d422540b02f5a3ca729a65921927cf557/locobot/agent/tasks.py#L346 , then it errors out.

I'll send a patch